multimodal/src/autogluon/multimodal/utils/metric.py

```diff
@@ -76,70 +79,93 @@ def infer_metrics(
     eval_metric_name
         Name of evaluation metric.
     """
+    is_customized = False
+    if eval_metric is None:
+        eval_metric_name = None
+    elif isinstance(eval_metric, str):
+        eval_metric_name = eval_metric
+    elif isinstance(eval_metric, Scorer):
+        eval_metric_name = eval_metric.name
+        is_customized = True
+    else:
+        raise TypeError(f"eval_metric can be a str, a Scorer, or None, but is type: {type(eval_metric)}")
+
+    if problem_type is not None:
+        problem_property = PROBLEM_TYPES_REG.get(problem_type)
+
+    if is_matching:
+        if eval_metric_name is not None:
+            # if eval_metric_name is a valid metric
+            if eval_metric_name.lower() in METRIC_MODE_MAP.keys():
+                validation_metric_name = eval_metric_name
+                return validation_metric_name, eval_metric_name
+            elif eval_metric_name.lower() in RETRIEVAL_METRICS:
+                # Currently only support recall as validation metric in retrieval tasks.
+                validation_metric_name = RECALL
+                return validation_metric_name, eval_metric_name
+            else:
+                if not is_customized:
+                    warnings.warn(
+                        f"Metric {eval_metric_name} is not supported as the evaluation metric for {problem_type} in matching tasks."
+                    )
+
+        # Fallback based on problem type unless it's a customized metric
+        if problem_type is None:
+            validation_metric_name, fallback_eval_metric_name = MATCHING_METRICS_WITHOUT_PROBLEM_TYPE
+            logger.info(
+                f"Metric {eval_metric_name} and metric {validation_metric_name} are used "
+                f"as the evaluation metric and the validation metric for matching tasks by default. "
+            )
+        elif problem_type in MATCHING_METRICS:
+            validation_metric_name, fallback_eval_metric_name = MATCHING_METRICS[problem_type]
+            logger.info(
+                f"Metric {eval_metric_name} and metric {validation_metric_name} are used "
+                f"as the evaluation metric and the validation metric for {problem_type} in matching tasks by default. "
+            )
+        else:
+            raise NotImplementedError(f"Problem type: {problem_type} is not yet supported for matching!")
+        if not is_customized:
+            eval_metric_name = fallback_eval_metric_name
```

Comment:
The logic looks weird. Since eval metric is not None when running there, no need to check is_customized. I think a better way is to just put `eval_metric_name` in the above returns instead of doing `eval_metric_name = fallback_eval_metric_name`

Reply:
At this point, the eval_metric_name is either None or invalid, so we fallback the metric if it's not a customized metric. We do want to fallback a customized metric.